src/components/freelancer/talent/skilldomainForm.tsx

```diff
@@ -16,56 +17,83 @@ import {
 } from '@/components/ui/table';
 import { axiosInstance } from '@/lib/axiosinstance';
 import { Switch } from '@/components/ui/switch';
+import { RootState } from '@/lib/store';
 
-// Define the types for skill and domain data
 interface Skill {
   label: string;
-  // Add other fields if necessary
 }
 
 interface Domain {
   label: string;
-  // Add other fields if necessary
 }
 
 interface SkillDomainData {
-  type: 'skill' | 'domain';
+  uid: string;
   label: string;
   experience: string;
   monthlyPay: string;
   status: string;
+  activeStatus: boolean;
 }
 
 const SkillDomainForm: React.FC = () => {
-  // Use the defined types for state variables
   const [skills, setSkills] = useState<Skill[]>([]);
   const [domains, setDomains] = useState<Domain[]>([]);
   const [skillDomainData, setSkillDomainData] = useState<SkillDomainData[]>([]);
   const [statusVisibility, setStatusVisibility] = useState<boolean[]>([]);
 
+  // Get the user data from Redux store
+  const user = useSelector((state: RootState) => state.user);
+
+  // Fetch skills, domains, and user's skill/domain data on mount
   useEffect(() => {
     async function fetchData() {
       try {
         const skillsResponse = await axiosInstance.get('/skills/all');
         setSkills(skillsResponse.data.data);
         const domainsResponse = await axiosInstance.get('/domain/all');
         setDomains(domainsResponse.data.data);
+
+        // Fetch the skill/domain data for the specific freelancer
+        if (user?.uid) {
+          const talentResponse = await axiosInstance.get(
+            `/freelancer/${user.uid}/dehix-talent`,
+          );
+          const talentData = talentResponse.data?.data[0]?.dehixTalent || {};
+
+          // Convert the talent object into an array
+          const formattedTalentData = Object.values(talentData).map(
+            (item: any) => ({
+              uid: item._id, // Ensure that the UID is present here
+              label: item.skillName || item.domainName || 'N/A',
+              experience: item.experience || 'N/A',
+              monthlyPay: item.monthlyPay || 'N/A',
+              status: item.status,
+              activeStatus: item.activeStatus,
+            }),
+          );
+
+          setSkillDomainData(formattedTalentData);
+          setStatusVisibility(
+            formattedTalentData.map((item) => item.activeStatus),
+          );
+        }
       } catch (error) {
         console.error('Error fetching data:', error);
       }
     }
     fetchData();
-  }, []);
+  }, [user?.uid]);
 
-  // Type the data parameter for onSubmitSkill and onSubmitDomain functions
+  // Handle skill/domain submission
   const onSubmitSkill = (data: {
     label: string;
     experience: string;
     monthlyPay: string;
   }) => {
     setSkillDomainData([
       ...skillDomainData,
-      { type: 'skill', ...data, status: 'Pending' },
+      { uid: `${Date.now()}`, ...data, status: 'pending', activeStatus: false }, // Generate a unique ID
```

Comment:
Date.now() is not a unique id, It might seem like a great idea to use it as uid but this will only be a unique id for when testing in local machine. There could easily be a case where 2 user trigger the submit button at exact same time which will fail one of the user's submission.
Use uuid library for generating actual uuid you can trust https://www.npmjs.com/package/uuid